src/quicktype-core/language/CPlusPlus.ts

```diff
@@ -762,7 +912,122 @@ inline ${optionalType}<T> get_optional(const json &j, const char *property) {
             });
         }
 
-        this.ensureBlankLine();
-        this.emitLine("#endif");
+        this.finishFile();
+    }
+
+    protected emitClassDefinition(c: ClassType, className: Name): void {
+        this.startFile(this.sourcelikeToString(className)+".hpp");
+        /** 
+         * Need to generate "imports", in terms 'c' has members, which
+         * are defined by others
+         */
+        let included: boolean = false;
+        this.forEachClassProperty(c, "none", (_name, _jsonName, property) => {
+
+            let p = property.type;
+
+            if (p instanceof ArrayType) {
```

Comment:
What if it's an array of arrays?  Same problem with `MapType`.  I think you need `directlyReachableTypes` in `TypeUtils.ts`.  Feel free to `export` it.  It basically recurs into a type's children until it hits a type for which your callback returns a `Set`, and then it gives you the union of all those sets.  So you'll want to return the type it gives you as a set if it's a type that you have an include file for, or return an empty set if it shouldn't recur any further.  The latter would be the case if the type it gives you is defined in the file you're currently emitting.